src/tls/certificate.rs

```diff
@@ -47,28 +47,19 @@ pub(crate) fn make_cert(keypair: &Keypair) -> Result<rcgen::Certificate, super::
         let mut buf =
             [0u8; LIBP2P_SIGNING_PREFIX_LENGTH + LIBP2P_SIGNATURE_ALGORITHM_PUBLIC_KEY_LENGTH];
         buf[..LIBP2P_SIGNING_PREFIX_LENGTH].copy_from_slice(&super::LIBP2P_SIGNING_PREFIX[..]);
-        buf[LIBP2P_SIGNING_PREFIX_LENGTH..].copy_from_slice(certif_pubkey);
+        buf[LIBP2P_SIGNING_PREFIX_LENGTH..].copy_from_slice(&certif_pubkey);
         keypair.sign(&buf)?
     };
 
     // Generate the libp2p-specific extension.
     let libp2p_extension: rcgen::CustomExtension = {
         let extension_content = {
             let serialized_pubkey = keypair.public().into_protobuf_encoding();
-            yasna::construct_der(|writer| {
-                writer.write_sequence(|writer| {
-                    writer
-                        .next()
-                        .write_bitvec_bytes(&serialized_pubkey, serialized_pubkey.len() * 8);
-                    writer
-                        .next()
-                        .write_bitvec_bytes(&libp2p_ext_signature, libp2p_ext_signature.len() * 8);
-                })
-            })
+            yasna::encode_der(&(serialized_pubkey, libp2p_ext_signature))
```

Comment:
What is the reason for this change?  I know this is more concise, but it is also much less clear that is correct.

Reply:
This is correct because I tested it on the go implementation. The main reason for this change is because it's more concise.

Reply:
same here, seems like go-libp2p is broken

Reply:
It is, but it is broken in the same way as the ECMAScript version, and this version also saves two bytes on the wire.  So I recommend matching its behavior.